src/TorchSharp/NN/Functional.cs

```diff
@@ -404,6 +404,50 @@ public static partial class functional
                     }
                 }
 
+                /// <summary>
+                /// Applies a 2D max pooling over an input signal composed of several input planes.
+                /// </summary>
+                /// <param name="input">The input tensor.</param>
+                /// <param name="kernelSize"></param>
+                /// <param name="stride"></param>
+                /// <param name="padding"></param>
+                /// <param name="dilation"></param>
+                /// <param name="ceil_mode"></param>
+                /// <returns></returns>
+                public static Tensor max_pool2d(Tensor input, long kernelSize, long? stride = null,
+                    long? padding = null, long? dilation = null, bool ceil_mode = false)
+                {
+                    return max_pool2d(input,
+                        new long[] { kernelSize },
+                        stride == null ? new long[] { 1, 1 } : new long [] { stride.Value, stride.Value },
```

Comment:
do you think we can stackalloc the arrays here too? or this is not a hot code path? same comment for `max_pool2d` too.